routing/unified_policy.go

```diff
@@ -0,0 +1,85 @@
+package routing
+
+import (
+	"github.com/go-errors/errors"
+	"github.com/lightningnetwork/lnd/channeldb"
+	"github.com/lightningnetwork/lnd/lnwire"
+)
+
+// unifiedPolicy represents a derived policy that covers a set of channels.
+type unifiedPolicy struct {
+	minHTLC lnwire.MilliSatoshi
+	maxHTLC lnwire.MilliSatoshi
+
+	edgePolicy
+}
+
+// update adds a new channel policy to the unified policy. This method can be
+// called on a nil pointer to initialize a unified policy struct. The maxHTLC
+// parameter is the best known upper bound for bandwidth. This can be an
+// accurate estimate, the channel capacity or, when no information is available,
+// lnwire.MaxMilliSatoshi.
+func (u *unifiedPolicy) update(maxHTLC lnwire.MilliSatoshi,
```

Comment:
should this also take the base payment amount as an input? if the amount being paid is greater than the channel capacity, do we need to include it's parameters in the GCD? we won't know how much will be added from fees, but may help to prune early

Reply:
In the new approach, the amt is used to prune.